jerry-core/ecma/builtin-objects/typedarray/ecma-builtin-typedarray-prototype.c

```diff
@@ -1052,9 +1052,9 @@ ecma_builtin_typedarray_prototype_join (ecma_value_t this_arg, /**< this argumen
   }
   ecma_free_value (separator_value);
 
+  ECMA_OP_TO_NUMBER_FINALIZE (length_number);
```

Comment:
Something is messed up here. There should be no `return` statements between `ECMA_OP_TO_NUMBER_TRY_CATCH` and `ECMA_OP_TO_NUMBER_FINALIZE`. IMHO this function needs a serious refactoring.

Reply:
Probably it's not the best practice to return between these macros, but in this case all early return statements contains the required `ecma_free_values` which generally present at the and of the function. I'd also vote for a refactor, but it may worth an other PR, since this PR's aim to fix the reported issue only.